linux/stat_hosttime/stat_hosttime.yml

```diff
@@ -16,34 +16,49 @@
   tasks:
     - name: "Test case block"
       block:
-        - include_tasks: ../setup/test_setup.yml
+        - name: "Test setup"
+          include_tasks: ../setup/test_setup.yml
           vars:
             skip_test_no_vmtools: true
 
-        - include_tasks: ../utils/get_vmware_toolbox_cmd_path.yml
+        - name: "Get VMware tools command path for guest OS"
+          include_tasks: ../utils/get_vmware_toolbox_cmd_path.yml
 
         - name: "Execute vmware-toolbox-cmd stat hosttime in guest OS"
           ansible.builtin.shell: "LC_ALL='en_US.UTF-8' TZ=UTC {{ vmware_toolbox_cmd_path }} stat hosttime"
-          register: stat_time
+          register: stat_time_output
           delegate_to: "{{ vm_guest_ip }}"
           poll: 0
 
-        # At the same time get ESXi host time by executing 'date' command
+        # At the same time, get ESXi host time by Ansible module community.vmware.vmware_host_facts
         - name: "Get real host time on {{ esxi_hostname }}"
-          ansible.builtin.shell: date -u "+%d %b %Y %H:%M:%S"
-          register: real_time
-          delegate_to: "{{ esxi_hostname }}"
-        - ansible.builtin.debug:
+          include_tasks: ../../common/esxi_get_host_facts.yml
+
+        - name: "Set the fact of host time got from {{ esxi_hostname }}"
+          ansible.builtin.set_fact:
+            real_time: "{{ ' '.join((host_facts.ansible_facts.host_date_time.date, host_facts.ansible_facts.host_date_time.time)) }}"
```

Comment:
For ensurance, here can add a default value or add "when" clauses in case there is variable not defined issue.